adapters/repos/db/vector/hnsw/search.go

```diff
@@ -226,28 +232,109 @@ func (h *hnsw) searchLayerByVectorWithDistancer(queryVector []float32,
 			continue
 		}
 
-		if len(candidateNode.connections[level]) > h.maximumConnectionsLayerZero {
-			// How is it possible that we could ever have more connections than the
-			// allowed maximum? It is not anymore, but there was a bug that allowed
-			// this to happen in versions prior to v1.12.0:
-			// https://github.com/weaviate/weaviate/issues/1868
-			//
-			// As a result the length of this slice is entirely unpredictable and we
-			// can no longer retrieve it from the pool. Instead we need to fallback
-			// to allocating a new slice.
-			//
-			// This was discovered as part of
-			// https://github.com/weaviate/weaviate/issues/1897
+		var slice *common.VectorUint64Slice
+		var slice2 *common.VectorUint64Slice
+		var slice3 *common.VectorUint64Slice
```

Comment:
IIUC we are keeping a reference to these 3 buffers so that we can put them back in the pool at the end of this loop cycle?

How about we allocate them outside of the loop (before L206), and reuse the same ones over and over until the loop is over, and then we put them back to the pool?
That might reduce pressure on memory, especially if the loop is tight and the pool can't reuse memory fast enough.